### PR TITLE
CYB-401 - Add a "No results" message to the Gallery

### PR DESCRIPTION
**Pull request by nicolaasmatthijs**
https://jira.ets.berkeley.edu/jira/browse/CYB-401

Add a "No results" message to the Gallery

### CODE REVIEW

**Comment by nicolaasmatthijs** [comment]
Rebased against master. Ready for review again

File: app/assets/templates/gallery_list.html [hunk 1]
```diff
@@ -45,3 +45,9 @@
     </div>
   </div>
 </div>
+<div data-ng-if="filteredItems.length === 0" class="alert alert-info dc-gallery-noresults">
+  <span data-ng-if="!sortAndFilter.selected.assignment && !sortAndFilter.selected.search.author">No Gallery Items are available yet</span>
+  <span data-ng-if="sortAndFilter.selected.assignment && !sortAndFilter.selected.search.author">No Gallery Items could be found for the assignment <strong>&quot;{{sortAndFilter.selected.assignment.name}}&quot;</strong></span>
```
Comment (christianv) [inline app/assets/templates/gallery_list.html]: @nicolaasmatthijs can we use `ng-bind` instead of `{{}}`

Comment (nicolaasmatthijs) [inline app/assets/templates/gallery_list.html]: The reduction in readability still makes me sad :(

**Comment by nicolaasmatthijs** [comment]
Followed up on code review. Back to @christianv for final check and merging

File: app/assets/templates/gallery_list.html [hunk 2]
```diff
@@ -45,3 +45,9 @@
     </div>
   </div>
 </div>
+<div data-ng-if="filteredItems.length === 0" class="alert alert-info dc-gallery-noresults">
+  <span data-ng-if="!sortAndFilter.selected.assignment && !sortAndFilter.selected.search.author">No Gallery Items are available yet</span>
+  <span data-ng-if="sortAndFilter.selected.assignment && !sortAndFilter.selected.search.author">No Gallery Items could be found for the assignment <strong>&quot;<span data-ng-bind="sortAndFilter.selected.assignment.name"></span>&quot;</strong></span>
+  <span data-ng-if="!sortAndFilter.selected.assignment && sortAndFilter.selected.search.author">No Gallery Items could be found for the student <strong>&quot;<span data-ng-bind="sortAndFilter.selected.search.author"></span>&quot;</strong></span>
+  <span data-ng-if="sortAndFilter.selected.assignment && sortAndFilter.selected.search.author">No Gallery Items could be found for the student <strong>&quot;<span data-ng-bind="sortAndFilter.selected.search.author"></span>&quot;</strong> in the assignment <strong><span data-ng-bind="sortAndFilter.selected.assignment.name"></span>&quot;&quot;</span>
```
Comment (christianv) [inline app/assets/templates/gallery_list.html]: @nicolaasmatthijs is it on intended to have a `&quot;&quot;` here? Or should the `sortAndFilter.selected.assignment.name` be between those quotes?

Comment (nicolaasmatthijs) [inline app/assets/templates/gallery_list.html]: Ugh, sorry. As you guessed, that wasn't the intent

**Comment by nicolaasmatthijs** [comment]
Fixed, Back ti @christianv for merging.

**Comment by christianv** [comment]
@nicolaasmatthijs lgtm, thanks!

**Pull request merged**